GVFS/GVFS.Mount/InProcessMount.cs

```diff
@@ -403,7 +437,20 @@ private void HandlePostIndexChangedRequest(NamedPipeMessages.Message message, Na
             }
             else
             {
-                this.fileSystemCallbacks.ForceIndexProjectionUpdate(request.UpdatedWorkingDirectory, request.UpdatedSkipWorktreeBits);
+                if (this.resetForDehydrateInProgress)
+                {
+                    // To avoid having to parse the index twice when dehydrating folders, repurpose the PostIndexChangedRequest
+                    // for git reset to rebuild the projection.  Additionally, if we were to call ForceIndexProjectionUpdate
+                    // directly in HandleDehydrateFolders we'd have a race condition where the OnIndexWriteRequiringModifiedPathsValidation
+                    // background task would be trying to parse the index at the same time as HandleDehydrateFolders
+
+                    this.fileSystemCallbacks.ForceIndexProjectionUpdate(invalidateProjection: true, invalidateModifiedPaths: false);

```

Comment:
Do we still need to pass `true` for the `invalidateModifiedPaths`?  `git reset` passes `true` for the `request.UpdatedSkipWorktreeBits` but in this case do we ever expect the reset that the dehydrate runs to change turn off skipworktree bits so those paths need to be added to the modified paths?  I don't think so since we are explictly passing in the paths that should be reset and they are the paths that were dehydrated and should get the skipworktree bits set.

Reply:
> but in this case do we ever expect the reset that the dehydrate runs to change turn off skipworktree bits so those paths need to be added to the modified paths? I don't think so since we are explictly passing in the paths that should be reset and they are the paths that were dehydrated and should get the skipworktree bits set.

Yeah that was my thought as well.  

In my first round of changes I didn't explicitly set `invalidateModifiedPaths` to false and `AddMissingModifiedFiles` was always getting called.  As you've pointed out, the `git reset` that we're running should never turn skip-worktree bits off as we're calling it for folders we just dehydrated.  With that in mind I made the switch to pass in `false` for perf reasons.  It shouldn't hurt anything functionality wise to leave `invalidateModifiedPaths` as `request.UpdatedSkipWorktreeBits` though, and so let me know if you'd prefer that.

Reply:
Nope this looks good to me.